### PR TITLE
Performance

### PR DESCRIPTION
**Pull request by FernandoES**
# Performance update in SettingsService

## Description

Avoidance of fast iterative settings update with system values and also of values copied to themselves.

Issue #63 

## Definition of Done

A task/pull request will not be considered to be complete until all these items can be checked off.

* [x] **All** requirements mentioned in the issue are implemented
* [x] Does match the Code of Conduct and the Contribution file 
* [x] Task has its own **GitHub issue** (something it is solving)
  * [x] Issue number is **included in the commit messages** for traceability
* [x] **Update the README.md** with any changes/additions made
* [x] **Update the CHANGELOG.md** with any changes/additions made
* [x] **Enough test coverage to ensure that uncovered changes do not break functionality**
* [x] **All tests pass**    
* [x] **Descriptive pull request text**, answering:
  + What problem/issue are you fixing?
  + What does this PR implement and how? 
* [x] **Assign your PR to someone** for a code review
  + This person _will be contacted **first**_ if a bug is introduced into `master`
* [x] **Manual testing** did not fail

### CODE REVIEW

File: visualization/app/codeCharta/core/settings/settings.service.ts [hunk 1]
```diff
@@ -294,12 +330,12 @@ export class SettingsService implements DataServiceSubscriber, CameraChangeSubsc
         this._settings.margin = settings.margin;
         this._settings.deltas = settings.deltas;
         this._settings.operation = settings.operation;
-        this._settings.deltaColorFlipped = this.settings.deltaColorFlipped;
+        this._settings.deltaColorFlipped = settings.deltaColorFlipped;
         this._settings.minimizeDetailPanel = settings.minimizeDetailPanel;
         this._settings.invertHeight = settings.invertHeight;
 
         //TODO what to do with map ? should it even be a part of settings ? deep copy of map ?
-        this._settings.map = this.settings.map;
```
Comment (mrpatpat) [inline visualization/app/codeCharta/core/settings/settings.service.ts]: using this.settings.map ignored the given map in the settings object. This allowed to set the settings without changing a map (used in scenarios! They do not provide a map). Setting a scenario prevents your implementation to work. Please adjust the scenario application to your methods.

Comment (FernandoES) [reply visualization/app/codeCharta/core/settings/settings.service.ts]: Thanks for the info, line 302 was corrected from 

        this._settings.map = settings.map;
to 
        this._settings.map = settings.map|| this.settings.map;

**Comment by FernandoES** [comment]
@mrpatpat In order to avoid waiting time in the ticks I changed their onClick value from triggering appySettings to onSettingsChanged.

Let's take into account that applySettings when called with no attributes waits 400ms and then, if not called again meanwhile, calls onSettingsChanged.

The change was done in the component or html which contained every tick of the panel.

**Comment by mrpatpat** [comment]
This feels better now. I was able to reproduce the bug mentioned before. Everything is noted in bug #163. This bug is not relevant to this issue since it also exists in master and must have been introduced before.

**Pull request merged**